app/src/main/java/com/rockspin/androiddevtest/MainActivity.java

```diff
@@ -1,18 +1,33 @@
 package com.rockspin.androiddevtest;
 
+import android.app.ProgressDialog;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
 import android.os.Bundle;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
+import android.view.View;
 
-import java.util.List;
+public class MainActivity extends AppCompatActivity implements View.OnClickListener
+{
 
-import retrofit2.Response;
+    private RecyclerView        mRecyclerView   = null;
```

Comment:
Every variable in a program must have a value before its value is used. I know that in JAVA the default value is null, but my c/c++ background affects me from time to time.